ext/opcache/Optimizer/sccp.c

```diff
@@ -1829,6 +1829,41 @@ static void sccp_visit_instr(scdf_ctx *scdf, zend_op *opline, zend_ssa_op *ssa_o
 			}
 			SET_RESULT_BOT(result);
 			break;
+		case ZEND_ECHO:
+			// fprintf(stderr, "optimizing ZEND_ECHO type=%d\n", Z_TYPE_P(op1));
+			SKIP_IF_TOP(op1);
+			if (IS_PARTIAL_ARRAY(op1)) {
+				break;
+			}
+			if (Z_TYPE_P(op1) == IS_STRING) {
+				/* already a string */
+			} else if (zend_optimizer_eval_cast(&zv, IS_STRING, op1) == SUCCESS) {
+				opline->op1_type = IS_CONST;
+				opline->op1.constant = zend_optimizer_add_literal(ctx->scdf.op_array, &zv);
```

Comment:
This is incorrect: The constant value here can be an optimistic assumption. It's not possible to modify instructions until SCCP has reached a fixed point.

Reply:
The right place to handle the conversion to string is inside https://github.com/php/php-src/blob/9f4d1b99362f4bf00b8ea27fca39af2faf6bb634/ext/opcache/Optimizer/zend_optimizer.c#L231.

Reply:
Done.